### ISSUE TITLE
Missing __truncdfsf2 for aarch64-unknown-none-softfloat

### ISSUE THREAD

**Issue by rahealy** [rust-lang/compiler-builtins#327]
I took a look at the C files referenced in this repo's README.md but don't feel that I'm qualified to make the port to rust.

I'd be very grateful if this function could be prioritized for implementation. Thank you.

**Comment by vinc** [comment]
I have the same issue it seems on a project targeting `86_64-unknown-none` and `-mmx,-sse,+soft-float` when using `libm::atan2`:

```
  = note: rust-lld: error: undefined symbol: __truncdfsf2
          >>> referenced by libm.3ij53uav-cgu.12
          >>>               libm-94520a1e125a28f6.libm.3ij53uav-cgu.12.rcgu.o:(libm::math::atan::atan::h43feb293179e50c4) in archive /home/v/src/vinc/moros/target/x86_64-moros/release/deps/liblibm-94520a1e125a28f6.rlib
```

**Comment by andre-richter** [comment]
There are two bare-metal aarch64 targets for rustc:

aarch64-unknown-none
aarch64-unknown-none-softfloat

The latter is intended for Kernels or Hypervisors where code that uses floating point would be concenptionally ill-suited in most cases.

It just happens that the current rust compiler infrastructure does fall back to fixed point implementations of float operations if avail, hence the unusual target name.

Hitting a case where a float intrinsic is missing with the softfloat  target is probably either a good hint that you accidentally included float code where you should not, or that the aarch64-unknown-none target is what you want instead.

**Comment by dbeckwith** [comment]
Ran into the same linker error on `x86_64-unknown-none` with the `soft-float` feature enabled. https://github.com/rust-num/num-traits/issues/167 has the details. I'll look into the implementation that was going to be added in #262 and see if that works for me.

**Comment by HTG-YT** [comment]
Just ran into this issue today whilst trying to target `x86_64-unknown-uefi`.

**Comment by sajattack** [comment]
Same thing on mips2 with `+single-float` (mipsel-sony-psp tier 3 target).

**Comment by sajattack** [comment]
Looks like it will be fixed soon? https://github.com/rust-lang/compiler-builtins/pull/448

**Comment by ayrtonm** [comment]
These should be available on nightly now that rust-lang/rust#91986 is merged.
